quickwit/quickwit-control-plane/src/control_plane_model.rs

```diff
@@ -245,7 +246,34 @@ impl ControlPlaneModel {
     }
 
     pub(crate) fn delete_source(&mut self, index_uid: &IndexUid, source_id: &SourceId) {
+        // Removing shards from shard table.
         self.shard_table.delete_source(index_uid, source_id);
+        // Remove source from index config.
+        let Some(index_model) = self.index_table.get_mut(index_uid) else {
+            warn!(index_uid=%index_uid, source_id=%source_id, "delete source: index not found");
+            return;
+        };
+        if index_model.sources.remove(source_id).is_none() {
+            warn!(index_uid=%index_uid, source_id=%source_id, "delete source: source not found");
+        };
+    }
+
+    /// Returns `true` if the source status has changed, `false` otherwise.
+    /// Returns an error if the source could not be found.
+    pub(crate) fn toggle_source(
+        &mut self,
+        index_uid: &IndexUid,
+        source_id: &SourceId,
+        enable: bool,
+    ) -> anyhow::Result<bool> {
+        let Some(index_model) = self.index_table.get_mut(index_uid) else {
+            bail!("Index `{}` not found.", index_uid);
+        };
+        let Some(source_config) = index_model.sources.get_mut(source_id) else {
+            bail!("Source `{}` not found.", source_id);
+        };
+        source_config.enabled = enable;
+        Ok(enable)
```

Comment:
This does not return whether the source has changed but rather whether it was enabled or not.

Reply:
that's a bug indeed.